src/appleseed.bench/mainwindow/rendering/renderingmanager.h

```diff
@@ -0,0 +1,137 @@
+
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2010-2013 Francois Beaune, Jupiter Jazz Limited
+// Copyright (c) 2014-2018 Francois Beaune, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#pragma once
+
+// appleseed.bench headers.
+#include "mainwindow/rendering/qtrenderercontroller.h"
+
+// appleseed.renderer headers.
+#include "renderer/api/rendering.h"
+#include "renderer/api/types.h"
+#include "renderer/api/utility.h"
+
+// appleseed.foundation headers.
+#include "foundation/math/transform.h"
+#include "foundation/platform/thread.h"
+#include "foundation/utility/autoreleaseptr.h"
+#include "foundation/utility/job/abortswitch.h"
+#include "foundation/utility/searchpaths.h"
+
+// Qt headers.
+#include <QObject>
+#include <QtGlobal>
+#include <QThread>
+
+// Standard headers.
+#include <cstdint>
+#include <memory>
+
+// Forward declarations.
+namespace appleseed     { namespace bench { class RenderWidget; } }
+namespace appleseed     { namespace bench { class RenderingTimeDisplay; } }
+namespace renderer      { class Project; }
+
+namespace appleseed {
+namespace bench {
+
+class RenderingManager
```

Comment:
Is it duplicated code ?

Reply:
Yes, there's a bit of duplication. It's not worth trying to factorize this code IMHO.